src/test/regress/sql/update_gp.sql

```diff
@@ -117,6 +117,184 @@ DROP TABLE keo3;
 DROP TABLE keo4;
 DROP TABLE keo5;
 
+-- Explicit Redistribute Motion should be added only if there is a motion
+-- between the scan and the ModifyTable on the relation we are going to modify.
+-- (test case not applicable to ORCA)
+-- start_ignore
+DROP TABLE IF EXISTS t1;
+DROP TABLE IF EXISTS t2;
+DROP TABLE IF EXISTS t_strewn;
+DROP TABLE IF EXISTS t_strewn2;
+-- end_ignore
+
+CREATE TABLE t1 (i int, j int) DISTRIBUTED BY (i);
+CREATE TABLE t2 (i int) DISTRIBUTED BY (i);
+CREATE TABLE t_strewn (i int) DISTRIBUTED RANDOMLY;
+CREATE TABLE t_strewn2 (i int) DISTRIBUTED RANDOMLY;
+
+EXPLAIN (costs off)
+UPDATE t1 SET j = t_strewn.i FROM t_strewn WHERE t_strewn.i = t1.i;
+
+EXPLAIN (costs off)
+WITH CTE AS (DELETE FROM t1 RETURNING *)
+SELECT count(*) AS a FROM t_strewn JOIN t2 USING (i)
+WHERE 0 < ALL (SELECT i FROM cte);
```

Comment:
What is being tested here? There is no complicated logic under the Modify Table node.

Reply:
It's a query from the ticket. Now I simplified it. Previosly, explicit redistribute motion was added under ModifyTable node, even if a redistribute/broadcast motion was above the delete.